mapie/mondrian.py

```diff
@@ -0,0 +1,438 @@
+    input a MapieClassifier, MapieRegressor or MapieMultiLabelClassifier
+    estimator and fits a model for each group of individuals. The Mondrian
+    class can then be used to run a conformal prediction procedure for each
+    of these groups and hence achieve marginal coverage on each of them.
+
+    Parameters
+    ----------
+    mapie_estimator : Union[MapieClassifier, MapieRegressor,
+                            MapieMultiLabelClassifier]
+        The estimator for which the Mondrian method will be applied. The
+        estimator must be used with cv='prefit' and the conformity score must
+        be one of the following:
+        - For MapieClassifier: 'lac', 'score', 'cumulated_score',
+        'aps' or 'topk'
+        - For MapieRegressor: 'gamma', 'absolute' or 'aps'
+
+    Attributes
+    ----------
+    unique_groups : NDArray
+        The unique groups of individuals for which the estimator was fitted
+    mapie_estimators : Dict
+        A dictionary containing the fitted conformal estimator for each group
+        of individuals
+
+    References
+    ----------
+    Vladimir Vovk, David Lindsay, Ilia Nouretdinov, and Alex Gammerman.
+    Mondrian confidence machine.
+    Technical report, Royal Holloway University of London, 2003
+
+    Examples
+    --------
+        >>> import numpy as np
+    >>> from sklearn.linear_model import LogisticRegression
+    >>> from mapie.classification import MapieClassifier
+    >>> X_toy = np.arange(9).reshape(-1, 1)
+    >>> y_toy = np.stack([0, 0, 1, 0, 1, 2, 1, 2, 2])
+    >>> groups = [0, 0, 0, 0, 1, 1, 1, 1, 1]
+    >>> clf = LogisticRegression(random_state=42).fit(X_toy, y_toy)
+    >>> mapie = Mondrian(MapieClassifier(estimator=clf, cv="prefit")).fit(
+    ...     X_toy, y_toy, groups)
+    >>> _, y_pi_mapie = mapie.predict(X_toy, alpha=0.4, groups=groups)
+    >>> print(y_pi_mapie[:, :, 0].astype(bool))
+    [[ True False False]
+     [ True False False]
+     [ True  True False]
+     [ True  True False]
+     [False  True False]
+     [False  True  True]
+     [False False  True]
+     [False False  True]
+     [False False  True]]
+    """
+
+    not_allowed_estimators = (
+        MapieQuantileRegressor,
+        MapieTimeSeriesRegressor
+    )
+    allowed_classification_ncs_str = [
+        "lac", "score", "cumulated_score", "aps", "top_k"
+    ]
+    allowed_classification_ncs_class = (
+        LACConformityScore, NaiveConformityScore, APSConformityScore,
+        TopKConformityScore
+    )
+    allowed_regression_ncs = (
+        GammaConformityScore, AbsoluteConformityScore, APSConformityScore
+    )
+    fit_attributes = [
+        "unique_groups",
+        "mapie_estimators"
+    ]
+
+    def __init__(
+        self, mapie_estimator: Union[
+            MapieCalibrator,
+            MapieClassifier,
+            MapieRegressor,
+            MapieMultiLabelClassifier
+        ]
+    ):
+        self.mapie_estimator = mapie_estimator
+
+    def _check_mapie_classifier(self):
```

Comment:
Merge this method with `_check_estimator` ?

Reply:
Meeting note: do not support multi label classif for Mondrian for now (cv param missing)